File: JavaScript/1-twoSum.js
var twoSum = (nums, target) => {
    var index = [];
    for(i = 0; i < nums.length; i++){
        for(j = i+1; j < (nums.length); j++) {
            total = nums[i] + nums[j]
            if(total == target) {
                index.push(i)
                index.push(j)
                return(index);
            }
        }
    }
}

console.log(twoSum([2,7,11,15], 9))
console.log(twoSum([3,2,4], 6))
console.log(twoSum([3,3], 6))

File: CS/1-TwoSum.cs
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        Dictionary<int, int> index = new Dictionary<int, int>();
        for(int i = 0; i < nums.Length; i++) {
            var result = target - nums[i];
            if(index.ContainsKey(result)) {
                return new int[]{index[result], i};
            } else {
                if(!index.ContainsKey(nums[i])){
                    index.Add(nums[i], i);
                }
            }
        }
        return new int[0];
    }
}